Installation failed due to "Failed building wheel for pgeof"

I am running into trouble installing the library using pip. It looks like an issue with the nanobind library. The last time I installed pgeof it went fine (a couple of months ago). The installation ends with:

> Failed to build pgeof
> ERROR: Could not build wheels for pgeof, which is required to install pyproject.toml-based projects


<details>
  <summary>Full log trace</summary>
python -m pip install pgeof
Collecting pgeof
  Downloading pgeof-0.2.0.tar.gz (69.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 69.0/69.0 MB 7.0 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy<2,>=1.7 (from pgeof)
  Using cached numpy-1.24.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.6 kB)
Using cached numpy-1.24.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
Building wheels for collected packages: pgeof
  Building wheel for pgeof (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pgeof (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [110 lines of output]
      *** scikit-build-core 0.9.4 using CMake 3.27.7 (wheel)
      *** Configuring CMake...
      2024-05-31 11:06:57,601 - scikit_build_core - WARNING - Can't find a Python library, got libdir=/home/matic/anaconda3/envs/test-env/lib, ldlibrary=libpython3.8.a, multiarch=x86_64-linux-gnu, masd=None
      loading initial cache file build/cp38-cp38-linux_x86_64/CMakeInit.txt
      -- The C compiler identification is GNU 9.4.0
      -- The CXX compiler identification is GNU 9.4.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Python: /home/matic/anaconda3/envs/test-env/bin/python (found suitable version "3.8.19", minimum required is "3.8") found components: Interpreter Development.Module
      -- Configuring done (0.4s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/build/cp38-cp38-linux_x86_64
      *** Building project with Ninja...
      [1/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_static_property.cpp.o
      [2/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/implicit.cpp.o
      [3/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/trampoline.cpp.o
      [4/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_internals.cpp.o
      [5/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/error.cpp.o
      [6/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_ndarray.cpp.o
      [7/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/common.cpp.o
      [8/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_enum.cpp.o
      [9/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_func.cpp.o
      [10/13] Building CXX object CMakeFiles/nanobind-static.dir/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/src/nb_type.cpp.o
      [11/13] Linking CXX static library libnanobind-static.a
      [12/13] Building CXX object CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o
      FAILED: CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o
      /usr/bin/c++ -Dpgeof_ext_EXPORTS -I/tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include -I/tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/third_party/eigen -I/tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/third_party/nanoflann/include -I/tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/third_party/taskflow -I/home/matic/anaconda3/envs/test-env/include/python3.8 -I/tmp/pip-build-env-34w23q9k/overlay/lib/python3.8/site-packages/nanobind/include -O3 -DNDEBUG -std=gnu++17 -flto -fno-fat-lto-objects -fPIC -fvisibility=hidden -fno-stack-protector -ffunction-sections -fdata-sections -MD -MT CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o -MF CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o.d -o CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o -c /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/src/pgeof_ext.cpp
      In file included from /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/src/pgeof_ext.cpp:8:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:76:58: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
         76 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>> compute_geometric_features(
            |                                                          ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:76:58: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:76:71: error: template argument 3 is invalid
         76 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>> compute_geometric_features(
            |                                                                       ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: In function ‘int pgeof::compute_geometric_features(pgeof::RefCloud<real_t>, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, size_t, bool)’:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:114:62: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        114 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>>(features, 2, shape, owner_features);
            |                                                              ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:114:62: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:114:75: error: template argument 3 is invalid
        114 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>>(features, 2, shape, owner_features);
            |                                                                           ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: At global scope:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:158:67: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        158 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any, feature_count>> compute_geometric_features_multiscale(
            |                                                                   ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:158:67: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:158:67: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:158:67: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:158:80: error: template argument 3 is invalid
        158 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any, feature_count>> compute_geometric_features_multiscale(
            |                                                                                ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: In function ‘int pgeof::compute_geometric_features_multiscale(pgeof::RefCloud<real_t>, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, const std::vector<unsigned int>&, bool)’:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:206:71: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        206 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any, feature_count>>(
            |                                                                       ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:206:71: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:206:71: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:206:71: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:206:84: error: template argument 3 is invalid
        206 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any, feature_count>>(
            |                                                                                    ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: At global scope:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:241:58: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        241 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>> compute_geometric_features_optimal(
            |                                                          ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:241:58: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:241:71: error: template argument 3 is invalid
        241 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>> compute_geometric_features_optimal(
            |                                                                       ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: In function ‘int pgeof::compute_geometric_features_optimal(pgeof::RefCloud<real_t>, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, nanobind::ndarray<const unsigned int, nanobind::shape<-1> >, uint32_t, uint32_t, uint32_t, bool)’:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:303:62: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        303 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>>(features, 2, shape, owner_features);
            |                                                              ^~~~~~~~~~~~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:303:62: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:303:75: error: template argument 3 is invalid
        303 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, feature_count>>(features, 2, shape, owner_features);
            |                                                                           ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: At global scope:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:320:62: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        320 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>> compute_geometric_features_selected(
            |                                                              ^~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:320:62: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:320:62: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:320:62: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:320:65: error: template argument 3 is invalid
        320 | static nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>> compute_geometric_features_selected(
            |                                                                 ^~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp: In function ‘int pgeof::compute_geometric_features_selected(pgeof::RefCloud<real_t>, real_t, uint32_t, const std::vector<pgeof::EFeatureID>&)’:
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:367:66: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
        367 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>>(features, {static_cast<size_t>(n_points), feature_count}, owner_features);
            |                                                                  ^~~
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:367:66: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:367:66: error: type/value mismatch at argument 1 in template parameter list for ‘template<long int ...Is> struct nanobind::shape’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:367:66: note:   expected a constant of type ‘long int’, got ‘nanobind::any’
      /tmp/pip-install-i_isslon/pgeof_16416ab81e6a4431b1b2dc0f09770a72/include/pgeof.hpp:367:69: error: template argument 3 is invalid
        367 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>>(features, {static_cast<size_t>(n_points), feature_count}, owner_features);
            |                                                                     ^~
      ninja: build stopped: subcommand failed.

      *** CMake build failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pgeof
Failed to build pgeof
ERROR: Could not build wheels for pgeof, which is required to install pyproject.toml-based projects

</details>

I tried installing inside a fresh conda environment using python3.8 and 3.9, both unsuccessful. Installation using [install.sh](https://github.com/drprojects/superpoint_transformer/blob/bb31689b1bf76d62112dc4d2094cb82dae3773ac/install.sh) inside the SPT repo also failed with the same error. Googling I found it could also be an issue with non-upgraded pip, wheel, but trying that didn't fix my issue.

I am using Ubuntu 20.04.4 LTS, CUDA Version 12.4

<details>
  <summary>Conda list of the fresh python3.9 conda environment</summary>

# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
ld_impl_linux-64          2.40                 hf3520f5_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h77fa898_7    conda-forge
libgomp                   13.2.0               h77fa898_7    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.45.3               h2797004_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.2.13               h4ab18f5_6    conda-forge
ncurses                   6.5                  h59595ed_0    conda-forge
openssl                   3.3.0                h4ab18f5_3    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
python                    3.9.19          h0755675_0_cpython    conda-forge
readline                  8.2                  h8228510_1    conda-forge
setuptools                70.0.0             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge

</details>

Any help is appreciated!

Hi @drprojects ,
I also have the same issue when installing point geometric features in superpoint transformer. 

Collecting git+https://github.com/drprojects/point_geometric_features.git
  Cloning https://github.com/drprojects/point_geometric_features.git to /tmp/pip-req-build-x05q6kos
  Running command git clone --filter=blob:none --quiet https://github.com/drprojects/point_geometric_features.git /tmp/pip-req-build-x05q6kos
  Resolved https://github.com/drprojects/point_geometric_features.git to commit 255bbc014f6e2b3d11d34dc149bdc5bd5ec3734f
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy<2,>=1.7 in /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages (from pgeof==0.2.0) (1.22.4)
Building wheels for collected packages: pgeof
  Building wheel for pgeof (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pgeof (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [279 lines of output]
      *** scikit-build-core 0.9.5 using CMake 3.29.3 (wheel)
      *** Configuring CMake...
      2024-06-02 10:12:37,196 - scikit_build_core - WARNING - Can't find a Python library, got libdir=/home/vaibhav/conda3v2/envs/superpoint/lib, ldlibrary=libpython3.8.a, multiarch=x86_64-linux-gnu, masd=None
      loading initial cache file build/cp38-cp38-linux_x86_64/CMakeInit.txt
      -- The C compiler identification is GNU 9.4.0
      -- The CXX compiler identification is GNU 9.4.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Python: /home/vaibhav/conda3v2/envs/superpoint/bin/python3.8 (found suitable version "3.8.19", minimum required is "3.8") found components: Interpreter Development.Module
      -- Configuring done (0.7s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/pip-req-build-x05q6kos/build/cp38-cp38-linux_x86_64
      *** Building project with Ninja...
      [1/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/trampoline.cpp.o
      [2/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/implicit.cpp.o
      [3/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/nb_enum.cpp.o
      [4/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/nb_internals.cpp.o
      [5/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/error.cpp.o
      [6/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/nb_ndarray.cpp.o
      [7/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/common.cpp.o
      [8/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/nb_func.cpp.o
      [9/12] Building CXX object CMakeFiles/nanobind-static.dir/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/src/nb_type.cpp.o
      [10/12] Linking CXX static library libnanobind-static.a
      [11/12] Building CXX object CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o
      FAILED: CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o
      /usr/bin/c++ -Dpgeof_ext_EXPORTS -I/tmp/pip-req-build-x05q6kos/include -I/tmp/pip-req-build-x05q6kos/third_party/eigen -I/tmp/pip-req-build-x05q6kos/third_party/nanoflann/include -I/tmp/pip-req-build-x05q6kos/third_party/taskflow -I/home/vaibhav/conda3v2/envs/superpoint/include/python3.8 -I/home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include -O3 -DNDEBUG -std=gnu++17 -flto -fno-fat-lto-objects -fPIC -fvisibility=hidden -fno-stack-protector -MD -MT CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o -MF CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o.d -o CMakeFiles/pgeof_ext.dir/src/pgeof_ext.cpp.o -c /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:7:
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:32:55: error: ‘ndim’ is not a member of ‘nb’
         32 | static std::pair<nb::ndarray<nb::numpy, uint32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                       ^~~~
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:32:60: error: template argument 3 is invalid
         32 | static std::pair<nb::ndarray<nb::numpy, uint32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                            ^
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:32:61: error: wrong number of template arguments (1, should be 2)
         32 | static std::pair<nb::ndarray<nb::numpy, uint32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                             ^~
      In file included from /usr/include/c++/9/bits/stl_algobase.h:64,
                       from /usr/include/c++/9/bits/char_traits.h:39,
                       from /usr/include/c++/9/string:40,
                       from /usr/include/c++/9/stdexcept:39,
                       from /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nanobind.h:31,
                       from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:2:
      /usr/include/c++/9/bits/stl_pair.h:208:12: note: provided for ‘template<class _T1, class _T2> struct std::pair’
        208 |     struct pair
            |            ^~~~
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:7:
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:32:63: error: expected unqualified-id before ‘,’ token
         32 | static std::pair<nb::ndarray<nb::numpy, uint32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                               ^
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:86:54: error: ‘ndim’ is not a member of ‘nb’
         86 | static std::pair<nb::ndarray<nb::numpy, int32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                      ^~~~
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:86:59: error: template argument 3 is invalid
         86 | static std::pair<nb::ndarray<nb::numpy, int32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                           ^
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:86:60: error: wrong number of template arguments (1, should be 2)
         86 | static std::pair<nb::ndarray<nb::numpy, int32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                            ^~
      In file included from /usr/include/c++/9/bits/stl_algobase.h:64,
                       from /usr/include/c++/9/bits/char_traits.h:39,
                       from /usr/include/c++/9/string:40,
                       from /usr/include/c++/9/stdexcept:39,
                       from /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nanobind.h:31,
                       from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:2:
      /usr/include/c++/9/bits/stl_pair.h:208:12: note: provided for ‘template<class _T1, class _T2> struct std::pair’
        208 |     struct pair
            |            ^~~~
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:7:
      /tmp/pip-req-build-x05q6kos/include/nn_search.hpp:86:62: error: expected unqualified-id before ‘,’ token
         86 | static std::pair<nb::ndarray<nb::numpy, int32_t, nb::ndim<2>>, nb::ndarray<nb::numpy, real_t, nb::ndim<2>>>
            |                                                              ^
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:8:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:77:59: error: ‘ndim’ is not a member of ‘nb’
         77 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                           ^~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:77:64: error: template argument 2 is invalid
         77 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                ^
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:77:65: error: expected ‘,’ or ‘...’ before ‘>’ token
         77 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                 ^~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, feature_count> > pgeof::compute_geometric_features(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:80:9: error: ‘k_min’ was not declared in this scope
         80 |     if (k_min < 1) { throw std::invalid_argument("k_min should be > 1"); }
            |         ^~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:82:35: error: ‘nn_ptr’ was not declared in this scope
         82 |     const size_t    n_points    = nn_ptr.size() - 1;  // number of points is not determined by xyz
            |                                   ^~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:84:35: error: ‘nn’ was not declared in this scope; did you mean ‘nb’?
         84 |     const uint32_t* nn_data     = nn.data();
            |                                   ^~
            |                                   nb
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In lambda function:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:96:17: error: ‘verbose’ was not declared in this scope
         96 |             if (verbose) log::progress(s_point, n_points);
            |                 ^~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:102:25: error: ‘k_min’ was not declared in this scope
        102 |             if (k_nn >= k_min)
            |                         ^~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, feature_count> > pgeof::compute_geometric_features(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:112:9: error: ‘verbose’ was not declared in this scope
        112 |     if (verbose) log::flush();
            |         ^~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: At global scope:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:159:59: error: ‘ndim’ is not a member of ‘nb’
        159 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                           ^~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:159:64: error: template argument 2 is invalid
        159 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                ^
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:159:65: error: expected ‘,’ or ‘...’ before ‘>’ token
        159 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                 ^~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, 18446744073709551615, feature_count> > pgeof::compute_geometric_features_multiscale(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:162:23: error: ‘k_scales’ was not declared in this scope; did you mean ‘check_scales’?
        162 |     if (!check_scales(k_scales))
            |                       ^~~~~~~~
            |                       check_scales
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:166:35: error: ‘nn_ptr’ was not declared in this scope
        166 |     const size_t    n_points    = nn_ptr.size() - 1;  // number of points is not determined by xyz
            |                                   ^~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:167:35: error: ‘k_scales’ was not declared in this scope; did you mean ‘n_scales’?
        167 |     const size_t    n_scales    = k_scales.size();
            |                                   ^~~~~~~~
            |                                   n_scales
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:169:35: error: ‘nn’ was not declared in this scope; did you mean ‘nb’?
        169 |     const uint32_t* nn_data     = nn.data();
            |                                   ^~
            |                                   nb
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In lambda function:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:182:17: error: ‘verbose’ was not declared in this scope
        182 |             if (verbose) log::progress(s_point, n_points);
            |                 ^~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, 18446744073709551615, feature_count> > pgeof::compute_geometric_features_multiscale(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:203:9: error: ‘verbose’ was not declared in this scope
        203 |     if (verbose) log::flush();
            |         ^~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: At global scope:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:242:59: error: ‘ndim’ is not a member of ‘nb’
        242 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                           ^~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:242:64: error: template argument 2 is invalid
        242 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                ^
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:242:65: error: expected ‘,’ or ‘...’ before ‘>’ token
        242 |     RefCloud<real_t> xyz, nb::ndarray<const uint32_t, nb::ndim<1>> nn, nb::ndarray<const uint32_t, nb::ndim<1>> nn_ptr,
            |                                                                 ^~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, feature_count> > pgeof::compute_geometric_features_optimal(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:245:9: error: ‘k_min’ was not declared in this scope
        245 |     if (k_min < 1 && k_min_search < 1) { throw std::invalid_argument("k_min and k_min_search should be > 1"); }
            |         ^~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:245:22: error: ‘k_min_search’ was not declared in this scope
        245 |     if (k_min < 1 && k_min_search < 1) { throw std::invalid_argument("k_min and k_min_search should be > 1"); }
            |                      ^~~~~~~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:247:35: error: ‘nn_ptr’ was not declared in this scope
        247 |     const size_t    n_points    = nn_ptr.size() - 1;  // number of points is not determined by xyz
            |                                   ^~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:249:35: error: ‘nn’ was not declared in this scope; did you mean ‘nb’?
        249 |     const uint32_t* nn_data     = nn.data();
            |                                   ^~
            |                                   nb
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In lambda function:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:261:17: error: ‘verbose’ was not declared in this scope
        261 |             if (verbose) log::progress(s_point, n_points);
            |                 ^~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:267:25: error: ‘k_min’ was not declared in this scope
        267 |             if (k_nn >= k_min && k_nn >= k_min_search)
            |                         ^~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:267:42: error: ‘k_min_search’ was not declared in this scope
        267 |             if (k_nn >= k_min && k_nn >= k_min_search)
            |                                          ^~~~~~~~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:278:42: error: ‘k_step’ was not declared in this scope
        278 |                     if ((k > k0) && (k % k_step != 0) && (k != k_nn)) { continue; }
            |                                          ^~~~~~
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In function ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, feature_count> > pgeof::compute_geometric_features_optimal(pgeof::RefCloud<real_t>, int)’:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:300:9: error: ‘verbose’ was not declared in this scope
        300 |     if (verbose) log::flush();
            |         ^~~~~~~
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp: In function ‘void nanobind_init_pgeof_ext(nanobind::module_&)’:
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:119:33: error: ‘nanoflann_knn_search’ is not a member of ‘pgeof’
        119 |     m.def("knn_search", &pgeof::nanoflann_knn_search<float>, "data"_a.noconvert(), "query"_a.noconvert(), "knn"_a, R"(
            |                                 ^~~~~~~~~~~~~~~~~~~~
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:119:54: error: expected primary-expression before ‘float’
        119 |     m.def("knn_search", &pgeof::nanoflann_knn_search<float>, "data"_a.noconvert(), "query"_a.noconvert(), "knn"_a, R"(
            |                                                      ^~~~~
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:132:34: error: ‘nanoflann_radius_search’ is not a member of ‘pgeof’
        132 |         "radius_search", &pgeof::nanoflann_radius_search<float>, "data"_a.noconvert(), "query"_a.noconvert(),
            |                                  ^~~~~~~~~~~~~~~~~~~~~~~
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:132:58: error: expected primary-expression before ‘float’
        132 |         "radius_search", &pgeof::nanoflann_radius_search<float>, "data"_a.noconvert(), "query"_a.noconvert(),
            |                                                          ^~~~~
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:8:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In instantiation of ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, 18446744073709551615> > pgeof::compute_geometric_features_selected(pgeof::RefCloud<real_t>, real_t, uint32_t, const std::vector<pgeof::EFeatureID>&) [with real_t = double; pgeof::RefCloud<real_t> = Eigen::Ref<const Eigen::Matrix<double, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >; typename Eigen::internal::conditional<const Eigen::Matrix<real_t, -1, 3, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type = Eigen::OuterStride<>; uint32_t = unsigned int]’:
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:163:11:   required from here
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:367:16: error: no matching function for call to ‘nanobind::ndarray<nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615> >::ndarray(double*&, <brace-enclosed initializer list>, nanobind::capsule&)’
        367 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>>(features, {static_cast<size_t>(n_points), feature_count}, owner_features);
            |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:3:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:273:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(nanobind::ndarray<Args>&&) [with Args = {nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        273 |     ndarray(ndarray &&t) noexcept : m_handle(t.m_handle), m_dltensor(t.m_dltensor) {
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:273:5: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:269:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(const nanobind::ndarray<Args>&) [with Args = {nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        269 |     ndarray(const ndarray &t) : m_handle(t.m_handle), m_dltensor(t.m_dltensor) {
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:269:5: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:251:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(void*, size_t, const size_t*, nanobind::handle, const int64_t*, nanobind::dlpack::dtype, int32_t, int32_t) [with Args = {nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615>}; size_t = long unsigned int; int64_t = long int; int32_t = int]’
        251 |     ndarray(void *value,
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:252:20: note:   no known conversion for argument 2 from ‘<brace-enclosed initializer list>’ to ‘size_t’ {aka ‘long unsigned int’}
        252 |             size_t ndim,
            |             ~~~~~~~^~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:246:14: note: candidate: ‘nanobind::ndarray<Args>::ndarray(nanobind::detail::ndarray_handle*) [with Args = {nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        246 |     explicit ndarray(detail::ndarray_handle *handle) : m_handle(handle) {
            |              ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:246:14: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:244:5: note: candidate: ‘constexpr nanobind::ndarray<Args>::ndarray() [with Args = {nanobind::numpy, double, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        244 |     ndarray() = default;
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:244:5: note:   candidate expects 0 arguments, 3 provided
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:8:
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp: In instantiation of ‘nanobind::ndarray<nanobind::numpy, real_t, nanobind::shape<18446744073709551615, 18446744073709551615> > pgeof::compute_geometric_features_selected(pgeof::RefCloud<real_t>, real_t, uint32_t, const std::vector<pgeof::EFeatureID>&) [with real_t = float; pgeof::RefCloud<real_t> = Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >; typename Eigen::internal::conditional<const Eigen::Matrix<real_t, -1, 3, 1>::IsVectorAtCompileTime, Eigen::InnerStride<1>, Eigen::OuterStride<> >::type = Eigen::OuterStride<>; uint32_t = unsigned int]’:
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:178:11:   required from here
      /tmp/pip-req-build-x05q6kos/include/pgeof.hpp:367:16: error: no matching function for call to ‘nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615> >::ndarray(float*&, <brace-enclosed initializer list>, nanobind::capsule&)’
        367 |     return nb::ndarray<nb::numpy, real_t, nb::shape<nb::any, nb::any>>(features, {static_cast<size_t>(n_points), feature_count}, owner_features);
            |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      In file included from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:3:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:273:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(nanobind::ndarray<Args>&&) [with Args = {nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        273 |     ndarray(ndarray &&t) noexcept : m_handle(t.m_handle), m_dltensor(t.m_dltensor) {
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:273:5: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:269:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(const nanobind::ndarray<Args>&) [with Args = {nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        269 |     ndarray(const ndarray &t) : m_handle(t.m_handle), m_dltensor(t.m_dltensor) {
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:269:5: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:251:5: note: candidate: ‘nanobind::ndarray<Args>::ndarray(void*, size_t, const size_t*, nanobind::handle, const int64_t*, nanobind::dlpack::dtype, int32_t, int32_t) [with Args = {nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615>}; size_t = long unsigned int; int64_t = long int; int32_t = int]’
        251 |     ndarray(void *value,
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:252:20: note:   no known conversion for argument 2 from ‘<brace-enclosed initializer list>’ to ‘size_t’ {aka ‘long unsigned int’}
        252 |             size_t ndim,
            |             ~~~~~~~^~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:246:14: note: candidate: ‘nanobind::ndarray<Args>::ndarray(nanobind::detail::ndarray_handle*) [with Args = {nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        246 |     explicit ndarray(detail::ndarray_handle *handle) : m_handle(handle) {
            |              ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:246:14: note:   candidate expects 1 argument, 3 provided
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:244:5: note: candidate: ‘constexpr nanobind::ndarray<Args>::ndarray() [with Args = {nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615>}]’
        244 |     ndarray() = default;
            |     ^~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/ndarray.h:244:5: note:   candidate expects 0 arguments, 3 provided
      In file included from /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nanobind.h:51,
                       from /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:2:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h: In instantiation of ‘PyObject* nanobind::detail::func_create(Func&&, Return (*)(Args ...), std::index_sequence<Is2 ...>, const Extra& ...) [with bool ReturnRef = false; bool CheckGuard = true; Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 11> > (*&)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 11> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; long unsigned int ...Is = {0, 1}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, char [1230]}; PyObject = _object; std::index_sequence<Is2 ...> = std::integer_sequence<long unsigned int, 0, 1>]’:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:161:37:   required from ‘void nanobind::cpp_function_def(Return (*)(Args ...), const Extra& ...) [with Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 11> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, char [1230]}]’
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:230:21:   required from ‘nanobind::module_& nanobind::module_::def(const char*, Func&&, const Extra& ...) [with Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 11> > (*)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Extra = {nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, char [1230]}]’
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:59:11:   required from here
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:49:39: error: static assertion failed: The number of nb::arg annotations must match the argument count!
         49 |     static_assert(nargs_provided == 0 || nargs_provided + is_method_det == nargs,
            |                   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h: In instantiation of ‘PyObject* nanobind::detail::func_create(Func&&, Return (*)(Args ...), std::index_sequence<Is2 ...>, const Extra& ...) [with bool ReturnRef = false; bool CheckGuard = true; Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615, 11> > (*&)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615, 11> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; long unsigned int ...Is = {0, 1}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, char [1278]}; PyObject = _object; std::index_sequence<Is2 ...> = std::integer_sequence<long unsigned int, 0, 1>]’:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:161:37:   required from ‘void nanobind::cpp_function_def(Return (*)(Args ...), const Extra& ...) [with Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615, 11> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, char [1278]}]’
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:230:21:   required from ‘nanobind::module_& nanobind::module_::def(const char*, Func&&, const Extra& ...) [with Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 18446744073709551615, 11> > (*)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Extra = {nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, char [1278]}]’
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:86:11:   required from here
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:49:39: error: static assertion failed: The number of nb::arg annotations must match the argument count!
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h: In instantiation of ‘PyObject* nanobind::detail::func_create(Func&&, Return (*)(Args ...), std::index_sequence<Is2 ...>, const Extra& ...) [with bool ReturnRef = false; bool CheckGuard = true; Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 12> > (*&)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 12> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; long unsigned int ...Is = {0, 1}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, char [1705]}; PyObject = _object; std::index_sequence<Is2 ...> = std::integer_sequence<long unsigned int, 0, 1>]’:
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:161:37:   required from ‘void nanobind::cpp_function_def(Return (*)(Args ...), const Extra& ...) [with Return = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 12> >; Args = {Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<-1> >, int}; Extra = {nanobind::scope, nanobind::name, nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, char [1705]}]’
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:230:21:   required from ‘nanobind::module_& nanobind::module_::def(const char*, Func&&, const Extra& ...) [with Func = nanobind::ndarray<nanobind::numpy, float, nanobind::shape<18446744073709551615, 12> > (*)(Eigen::Ref<const Eigen::Matrix<float, -1, 3, 1, -1, 3>, 0, Eigen::OuterStride<> >, int); Extra = {nanobind::arg, nanobind::arg, nanobind::arg, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, nanobind::arg_v, char [1705]}]’
      /tmp/pip-req-build-x05q6kos/src/pgeof_ext.cpp:118:11:   required from here
      /home/vaibhav/conda3v2/envs/superpoint/lib/python3.8/site-packages/nanobind/include/nanobind/nb_func.h:49:39: error: static assertion failed: The number of nb::arg annotations must match the argument count!
      ninja: build stopped: subcommand failed.
      
      *** CMake build failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pgeof
Failed to build pgeof
ERROR: Could not build wheels for pgeof, which is required to install pyproject.toml-based projects


I am using Ubuntu 20.04 and Cuda 11.8. Also, it is same case as @noisyneighbour. Last time it went fine, but this time, getting this issue.
Please have a look. Thankyou for considering.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["scikit-build-core >=0.4.3", "nanobind >= 1.9.2"]
+requires = ["scikit-build-core >=0.4.3", "nanobind == 1.9.2"]
 build-backend = "scikit_build_core.build"
 
 [project]

